AI/src/search/mcts/nodes/MP_PNMCTSNode.java

```diff
@@ -232,22 +235,30 @@ public boolean setProofAndDisproofNumbers()
 						{
 							// An unexpanded child
 							// TODO verify that this is indeed what we want to do?
-							proof = 1.0;
+							proof = Math.min(1.0, proof);
 						}
 					}
 
 	                if (this.proofNumbers[playerNum] != proof) 
 	                {
 						this.proofNumbers[playerNum] = proof;
 						
-						if (proof == 0.0) {
+						if (proof == 0.0) 
+						{
 							proofValue[playerNum] = MP_PNMCTSNodeValues.TRUE;
-							for(int p = 1; p <= numPlayers; p++)
-								if(p != playerNum)
+							for (int p = 1; p <= numPlayers; p++)
+							{
+								if (p != playerNum)
+								{
 									proofValue[p] = MP_PNMCTSNodeValues.FALSE;
+									proofNumbers[p] = Double.POSITIVE_INFINITY;
```

Comment:
forgetting this was the main bug. Setting the proof value to false meant we'd stop updated proof numbers, so it'd be a a disproven node, but proof number (used in selection phase) not correctly updated to infinity yet.